middlewares/middlewares.go

```diff
@@ -104,12 +108,33 @@ func AccessControl() negroni.Handler {
 
 // JwtMiddleware check if actual request have JWT
 func JwtMiddleware(key string, algo string) negroni.Handler {
-	jwtMiddleware := jwtmiddleware.New(jwtmiddleware.Options{
-		ValidationKeyGetter: func(token *jwt.Token) (interface{}, error) {
-			return []byte(key), nil
-		},
-		SigningMethod: jwt.GetSigningMethod(algo),
-	})
+	issuerURL, err := url.Parse("https://127.0.0.1/")
+	if err != nil {
+		log.Fatalf("Failed to parse the issuer url: %v", err)
+	}
+	provider := jwks.NewCachingProvider(issuerURL, 5*time.Minute)
+	customClaims := &auth.Claims{}
```

Comment:
The `validator.ValidatedClaims` already has the RegisteredClaims (jwt.StandardClaims from jwt-go pkg). So our custom claims should only be `auth.User`. 
```suggestion
	customClaims := &auth.User{}
```

We'll be able to access the RegisteredClaims through something like:

```
token := ctx.Request.Context().Value(jwtmiddleware.ContextKey{}).(*validator.ValidatedClaims)

claims := token.RegisteredClaims
```